packages/api/src/controllers/experiment/attestation.ts

```diff
@@ -26,22 +28,28 @@ const TYPES = {
 };
 
 app.post("/", validatePost("attestation"), async (req, res) => {
-  const { message, signature } = req.body;
+  const { message, signature, signatureType } = req.body;
 
-  if (!verifySigner(message, signature)) {
-    return res.status(400).json({ errors: ["invalid signature"] });
-  }
   if (!verifyTimestamp(message.timestamp)) {
     return res.status(400).json({
       errors: [
         "message timestamp must be in the present or past, not in the future",
       ],
     });
   }
+  const verfiedSignatureType = await verifySignature(
+    message,
+    signature,
+    signatureType
+  );
+  if (!verfiedSignatureType) {
+    return res.status(400).json({ errors: ["invalid signature"] });
+  }
 
-  const id = ethers.TypedDataEncoder.hash(DOMAIN, TYPES, message);
+  const id = uuid();
```

Comment:
Agreed on trying to use at least a hash here

Reply:
ok! You're both right. Updated to hash: cbdaccdf44ae7f2128424f1962dced96e2cc5886